Clean up the code

Indrre opened the pull request.

Indrre committed:
Clean up the code

Indrre committed:
Add RingView animation, timer and fast functionality

RichAppz commented on Fasting App/Extensions/Extensions.swift:
```diff
@@ -353,3 +353,54 @@ extension ActivityPickerView: UIPickerViewDataSource, UIPickerViewDelegate {
     }
 }
+
+extension UIViewController {
+    func setBackground() {
+        
+        let layer = CAGradientLayer()
+        guard
+            let mainColor = UIColor.stdBackground,
+            let topColor = UIColor.topBackground else { return }
+        
+        layer.frame = view.bounds
+        layer.colors = [mainColor.cgColor, topColor.cgColor]
+        layer.startPoint = CGPoint(x: 0, y: 0)
+        layer.endPoint = CGPoint(x: 1, y: 1)
+        view.layer.insertSublayer(layer, at: 0)
+    }
+}
```
Create extension files for each type of extension, for example

`UIViewController+Extensions.swift`

and this will contain all you're UIViewController extensions.

It is also good to comment on extensions functions to explain what they do.

Indrre replied:
Done

RichAppz commented on Fasting App/Extensions/Extensions.swift:
```diff
@@ -353,3 +353,54 @@ extension ActivityPickerView: UIPickerViewDataSource, UIPickerViewDelegate {
+        return String(format: "%02i:%02i:%02i", hours, minutes, seconds)
+    }
+    
+    var dayCharacter: String {
+        let date = Date(timeIntervalSince1970: self)
+        let day = Calendar.current.component(.weekday, from: date)
+        switch day {
+        case 1: return "S"
+        case 2: return "M"
+        case 3: return "T"
+        case 4: return "W"
+        case 5: return "T"
+        case 6: return "F"
+        case 7: return "S"
+        default: return ""
+        }
+    }
```
This process is not required.

Replace the switch with something like:

```swift 
var dayCharacter: String {
         let dateFormatter = DateFormatter()
         dateFormatter.dateFormat = "EEEE"
         let weekDay = dateFormatter.string(from: Date())
         return weekDay.first
 }

 //eg: // "EEEE" --> "Friday" 
       // "EEE"  --> "Fri"
```

RichAppz commented on Fasting App/Main/Controllers/MainViewController.swift:
```diff
@@ -121,16 +110,13 @@ class MainViewController: ViewController {
         super.viewDidAppear(animated)
         
-//        if let fast = FastStore.shared.currentFast {
-//            fastRingView.handleStart(fast: fast.start)
-//        }
-//        updateUI()
     }
```
Remove altogether if you are not using

Indrre replied:
Done

RichAppz commented on Fasting App/Main/View/ProfileHeaderView.swift:
```diff
@@ -136,11 +136,19 @@ class ProfileHeaderView: TouchableView {
         lblGreeting.text = timeOfTheDay
     }
 }
+
+extension String {
+    
+       static let goodMorning: String = "Good Morning"
+       static let goodAfternoon: String = "Good Afternoon"
+       static let goodEvening: String = "Good Evening"
+       static let defaultGreeting: String = "Hi"
+}
```
Put these in a `Constants` folder and in a `Strings.swift` file

RichAppz commented on Fasting App/Main/View/RingView.swift:
```diff
@@ -10,39 +10,43 @@ import UIKit
-    var timeSelected: Int?
+    let timer: String?
+    let fast: String?
+    var timeSelected: TimeInterval?
```
I can't see a `mutating func` so is there a reason why this is a var?

RichAppz commented on Fasting App/Main/View/RingView.swift:
```diff
@@ -10,39 +10,43 @@ import UIKit
+    let timeLapsed: Float
+    let stroke: CGFloat?
     let prsentPicker: (() -> Void)?
     var stopStartBtn: ((_ state: State) -> Void)?
```
And here

RichAppz commented on Fasting App/Main/ViewModel/MainViewModel.swift:
```diff
@@ -209,19 +213,41 @@ class MainViewModel {
+                self.stroke = Double(stroke)
+            }
+        }
+       
```
remove line

RichAppz commented on Fasting App/Main/ViewModel/MainViewModel.swift:
```diff
@@ -252,14 +278,15 @@ extension FastPickerView: UIPickerViewDataSource, UIPickerViewDelegate {
+        
         if component == 0 {
-           return 1 // hours header
+           return 1 // "Days" Header
         } else if component == 1 {
-           return 31 // hours
+           return 32 // Number of days
         } else if component == 2 {
-           return 1 // days header
+           return 1 // "Hours" Header
         } else {
-           return 25 // days
+           return 25 // Number of hours
         }
     }
```
This needs to use an enum, allowing for readability

RichAppz commented on Fasting App/Profile/ViewModel/ProfileSettingsViewModel.swift:
```diff
@@ -44,11 +44,11 @@ class ProfileSettingViewModel: NSObject, UIImagePickerControllerDelegate & UINav
-            lblGenderValue: user?.gender ?? "Female",
-            lblActivityValue: user?.activity ?? "Inactive ",
+            age: String(user?.age ?? 18),
+            weight: String(weight ?? "0.0") + (user?.weightUnit ?? "0.0"),
+            height: String(height  ?? "0") + (user?.heightMsureUnit ?? "0"),
```
Not sure why you have `String(weight ?? "0.0") + (user?.weightUnit ?? "0.0")`

What is meant to be displayed?

RichAppz commented on Fasting App/Service/FastService.swift:
```diff
@@ -68,6 +68,7 @@ class FastService {
             observers.forEach {
                 $0.observer?.fastServiceFastUpdated(currentFast)
             }
+            print("DEBUG: Timer - current fast \(currentFast)")
```
`debugPrint`, if you need to still have it stop using `print`

Indrre committed:
Add fast control, pickers and animation

Indrre committed:
Add FastView with Horizontal timer, bar view and history of fasts

Indrre committed:
Built Water and Weight views

Indrre committed:
before updating packages

Indrre committed:
Reset the package cache

Indrre committed:
add to app centre

Indrre committed:
add to app centre

Indrre committed:
xcscheme

Indrre committed:
return login

Indrre committed:
Fixed scrollView on the main screen
Add default pictures
Fix some constraints

Indrre committed:
Fixed main view scrollView
Fixed some constraints
Removed Calorie tile
Add default profile pictures

Indrre committed:
Fixed main view scrollView
Fixed some constraints
Removed Calorie tile
Add default profile pictures

Indrre committed:
fixed background gradient

Indrre committed:
Preload user profile pickers

Indrre committed:
Change profile pic size on the main screen
Reduce profile picture size
Remove Sign out button from scroll view and attached to the bottom of the screen

If water / weight is empty - open the picker

Preload Weight Picker for a new day
Fixed Ring bug - overlapping layer
Timer colour track colour change

Indrre committed:
Water, weight and Main tiles update on a new day
Other bug fixes

Indrre committed:
bug fix

Indrre committed:
bug fix

Indrre committed:
Final

Indrre committed:
enpty message

Indrre committed:
2.0 release

Indrre committed:
for PR request
